src/main/java/net/glasslauncher/annoyancefix/mixin/HatchetMixin.java

```diff
@@ -1,36 +1,49 @@
 package net.glasslauncher.annoyancefix.mixin;
 
+import com.google.common.collect.ObjectArrays;
 import net.minecraft.block.BlockBase;
 import net.minecraft.item.tool.Hatchet;
 import net.minecraft.item.tool.ToolBase;
 import net.minecraft.item.tool.ToolMaterial;
+import org.objectweb.asm.Opcodes;
 import org.spongepowered.asm.mixin.Mixin;
 import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
 
 @Mixin(Hatchet.class)
-public abstract class HatchetMixin extends ToolBase
-{
-    @Shadow private static BlockBase[] effectiveBlocks = new BlockBase[]
-            { BlockBase.WOOD
-            , BlockBase.BOOKSHELF
-            , BlockBase.LOG
-            , BlockBase.CHEST
-            , BlockBase.WORKBENCH
-            , BlockBase.NOTEBLOCK
-            , BlockBase.WOOD_DOOR
-            , BlockBase.LADDER
-            , BlockBase.STANDING_SIGN
-            , BlockBase.WALL_SIGN
-            , BlockBase.WOODEN_PRESSURE_PLATE
-            , BlockBase.JUKEBOX
-            , BlockBase.WOOD_STAIRS
-            , BlockBase.FENCE
-            , BlockBase.PUMPKIN
-            , BlockBase.JACK_O_LANTERN
-            , BlockBase.TRAPDOOR
-            };
+class HatchetMixin extends ToolBase {
+    @Shadow private static BlockBase[] effectiveBlocks;
 
-    protected HatchetMixin(int i, ToolMaterial arg) {
-        super(i, 3, arg, effectiveBlocks);
+    public HatchetMixin(int i, int j, ToolMaterial arg, BlockBase[] args) {
+        super(i, j, arg, args);
+    }
+
+    @Inject(
+            method = "<clinit>",
```

Comment:
Does "clinit" stand for "client init" and if so, does this code only run for the client? Would it work on servers?

Reply:
`<clinit>` is the static initializer, which is everything that happens under `static {}` block, including static fields initialization. It's not sided.